### ISSUE TITLE
EnumConverter from 16.2.0 doesn't work for the System.Environment.SpecialFolder enum 

### ISSUE THREAD

**Issue by quyenhm** [JoshClose/CsvHelper#1641]
This error occurs because the `SpecialFolder` enum has 2 members with the same value `5` is `Personal` and `MyDocuments`
So the dictionary `valueNames` that's being used in the `EnumConverter` class will get an exception when adding the second one.

https://github.com/JoshClose/CsvHelper/blob/74cb8d300a63632e498601c4a76dfb2ee9bf0189/src/CsvHelper/TypeConversion/EnumConverter.cs#L22
https://github.com/JoshClose/CsvHelper/blob/74cb8d300a63632e498601c4a76dfb2ee9bf0189/src/CsvHelper/TypeConversion/EnumConverter.cs#L52

Docs from Microsoft:
https://docs.microsoft.com/en-us/dotnet/api/system.environment.specialfolder

![image](https://user-images.githubusercontent.com/22790697/104127918-58357e80-5397-11eb-9ad2-8d6b5c59b7e9.png)

This is my code:
```cs
internal class Program
{
  public class Model
  {
    public Environment.SpecialFolder Special { get; set; }
  }
  private static void Main(string[] args)
  {
    string csvText = "Special\n" +
      "Desktop\n" +
      "Programs\n" +
      "Favorites\n" +
      "MyComputer\n" +
      "MyComputer\n";

    using (StringReader reader = new StringReader(csvText))
    using (CsvReader csvReader = new CsvReader(reader, CultureInfo.InvariantCulture))
    {
      List<Model> records = csvReader.GetRecords<Model>().ToList();
    }
  }
}
```

**Comment by JoshClose** [comment]
It looks like `Enum.Parse` will use the first one in the list, so that's what I'll do also.

**Comment by JoshClose** [comment]
I have it handling duplicate names and values for attributes the same way `Enum.TryParse` does and falling back to using `TryParse` if there is no attribute. This makes it easier to handle all the other cases.

I'll have a release for this soon.

**Comment by JoshClose** [comment]
Fixed in 22.0.0 on NuGet. https://www.nuget.org/packages/CsvHelper/22.0.0
